hnix-store-remote/src/System/Nix/Store/Remote/Util.hs

```diff
@@ -31,36 +48,102 @@ genericIncremental getsome parser = go decoder
           error msg
 
 getSocketIncremental :: Get a -> MonadStore a
-getSocketIncremental = genericIncremental sockGet
+getSocketIncremental = genericIncremental sockGet8
+  where
+    sockGet8 :: MonadStore (Maybe BSC.ByteString)
+    sockGet8 = do
+      soc <- storeSocket <$> ask
+      liftIO $ Just <$> recv soc 8
 
 sockPut :: Put -> MonadStore ()
 sockPut p = do
-  soc <- ask
-  liftIO $ sendAll soc $ LBS.toStrict $ runPut p
+  soc <- storeSocket <$> ask
+  liftIO $ sendAll soc $ BSL.toStrict $ runPut p
 
-sockGet :: MonadStore (Maybe BSC.ByteString)
-sockGet = do
-  soc <- ask
-  liftIO $ Just <$> recv soc 8
+sockGet :: Get a -> MonadStore a
+sockGet = getSocketIncremental
 
 sockGetInt :: Integral a => MonadStore a
 sockGetInt = getSocketIncremental getInt
 
 sockGetBool :: MonadStore Bool
 sockGetBool = (== (1 :: Int)) <$> sockGetInt
 
-sockGetStr :: MonadStore LBS.ByteString
+sockGetStr :: MonadStore BSL.ByteString
 sockGetStr = getSocketIncremental getByteStringLen
 
-sockGetStrings :: MonadStore [LBS.ByteString]
+sockGetStrings :: MonadStore [BSL.ByteString]
 sockGetStrings = getSocketIncremental getByteStrings
 
-lBSToText :: LBS.ByteString -> Text
-lBSToText = T.pack . BSC.unpack . LBS.toStrict
+sockGetPath :: MonadStore StorePath
+sockGetPath = do
+  sd <- getStoreDir
+  pth <- getSocketIncremental (getPath sd)
+  case pth of
+    Left e -> throwError e
+    Right x -> return x
+
+sockGetPathMay :: MonadStore (Maybe StorePath)
+sockGetPathMay = do
+  sd <- getStoreDir
+  pth <- getSocketIncremental (getPath sd)
+  return $ case pth of
+    Left e -> Nothing
+    Right x -> Just x
+
+sockGetPaths :: MonadStore StorePathSet
+sockGetPaths = do
+  sd <- getStoreDir
+  getSocketIncremental (getPaths sd)
+
+lBSToText :: BSL.ByteString -> Text
+lBSToText = T.pack . BSC.unpack . BSL.toStrict
```

Comment:
lBSToText seems weird. Why not BSLToText ?

Reply:
Renamed to `bslToText`, most stuff uses strict bytestrings now.